src/components/templates/OrderTemplate.tsx

```diff
@@ -0,0 +1,145 @@
+import { queryCart } from '@api/cartApi';
+import { CartProductData, CartedOptionData } from '@api/dto';
+import { order } from '@api/orderApi';
+import Card from '@components/atoms/Card';
+import InnerFlatCard from '@components/atoms/InnerFlatCard';
+import Checkbox from '@components/atoms/Checkbox';
+import comma from '@utils/commaUtils';
+import React, { useState, useRef } from 'react';
+import { useMutation, useQuery } from 'react-query';
+import { useNavigate } from 'react-router-dom';
+import FilledButton from '@components/atoms/button/FilledButton';
+import OrderedOption from '@components/atoms/OrderedOption';
+
+const OrderTemplate = () => {
+  const { data, error, isLoading } = useQuery('cart', queryCart);
+
+  const [allAgree, setAllAgree] = useState(false);
+  const [agreement, setAgreement] = useState<{ [key: string]: boolean }>({ agreePayment: false, agreePolicy: false });
+
+  const allAgreeRef = useRef(null);
+  const agreePaymentRef = useRef(null);
+  const agreePolicyRef = useRef(null);
+
+  const { products, totalPrice } = data?.data?.response ?? { products: undefined, totalPriace: undefined };
+  const { mutate: orderProduct } = useMutation({
+    mutationFn: () => order(),
+  });
+  const navigate = useNavigate();
+
+  const handleAllAgree = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target?.checked;
+    setAllAgree(value);
+    Object.keys(agreement).forEach((key) => {
+      agreement[key] = value;
+    });
+  };
+
+  const handleAgreement = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, checked } = e.target;
+
+    if (name === 'agreePayment') {
```

Comment:
key값으로 매칭해서 상태변경을 하면 if문을 줄일 수 있어보이네요
```suggestion
  setAgreement((prevAgreement) => ({ ...prevAgreement, [name]: checked }));
```